demos/python_demos/common/performance_metrics.py

```diff
@@ -0,0 +1,96 @@
+"""
+ Copyright (C) 2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from time import perf_counter
+import cv2
+from helpers import put_highlighted_text
+
+
+class Statistic:
+    def __init__(self):
+        self.latency = 0.0
+        self.period = 0.0
+        self.frame_count = 0
+
+    def combine(self, other):
+        self.latency += other.latency
+        self.period += other.period
+        self.frame_count += other.frame_count
+
+
+class PerformanceMetrics:
+    def __init__(self, time_window=1.0):
+        # 'time_window' defines the length of the timespan over which the 'current fps' value is calculated
+        self.time_window_size = time_window
+        self.last_moving_statistic = Statistic()
+        self.current_moving_statistic = Statistic()
+        self.total_statistic = Statistic()
+        self.last_update_time = None
+    
+    def update(self, last_request_start_time, frame, position=(15, 30),
+               font_scale=0.75, color=(200, 10, 10), thickness=2):
+        current_time = perf_counter()
+
+        last_request_latency = current_time - last_request_start_time
+        if self.total_statistic.frame_count != 0:
+            self.current_moving_statistic.latency += last_request_latency
+            self.current_moving_statistic.period = current_time - self.last_update_time
+            self.current_moving_statistic.frame_count += 1
+        else:
+            # for the 1st frame
+            self.last_update_time = current_time
+            self.last_moving_statistic.latency = last_request_latency
+            self.last_moving_statistic.frame_count = 1
+            self.total_statistic.latency = last_request_latency
+            self.total_statistic.frame_count = 1
+
+        if current_time - self.last_update_time > self.time_window_size:
```

Comment:
I think this `if` should be moved to the true branch of the preceding `if`, since this cannot happen on the first frame.